lookout/style/format/langs/javascript/postprocessor.py

```diff
@@ -21,32 +50,42 @@ def postprocess(X: numpy.ndarray, y: numpy.ndarray, vnodes_y: Sequence[VirtualNo
     :param vnodes: Sequence of virtual nodes representing the input.
     :param winners: Indices of the rules that were used to compute the predictions.
     :param rules: Rules used to perform the prediction.
+    :param feature_extractor: FeatureExtractor used to extract features.
     :return: Updated y and winners.
     """
     quotes_classes = {CLASS_INDEX[CLS_DOUBLE_QUOTE], CLASS_INDEX[CLS_SINGLE_QUOTE]}
     processed_y = y.copy()
     processed_winners = winners.copy()
-    vnodes_y_set = set(id(vnode) for vnode in vnodes_y)
-    y_indices = {}
-    for i, vnode in enumerate(vnodes):
-        if id(vnode) in vnodes_y_set:
-            y_indices[i] = len(y_indices)
-    for i, y_i in y_indices.items():
-        if not i + 2 in y_indices:
-            continue
-        y_i_paired = y_indices[i + 2]
-        vnode, vnode_in_between, vnode_paired = vnodes[i], vnodes[i + 1], vnodes[i + 2]
-        if vnode.y not in quotes_classes or vnode_paired.y not in quotes_classes:
+    y_indices = {id(vnode): i for i, vnode in enumerate(vnodes_y)}
+    new_rules = {}
+    for vnode1, vnode2, vnode3 in zip(vnodes, islice(vnodes, 1, None), islice(vnodes, 2, None)):
+        if (id(vnode1) not in y_indices or id(vnode3) not in y_indices or vnode2.node is None
+                or vnode1.y[-1] not in quotes_classes or vnode3.y[0] != vnode1.y[-1]):
             continue
-        if vnode_in_between.node is None:
+        vnode2_roles = frozenset(role_name(role_id) for role_id in vnode2.node.roles)
+        if "STRING" not in vnode2_roles:
             continue
-        vnode_in_between_roles = {role_name(role_id) for role_id in vnode_in_between.node.roles}
-        if "STRING" in vnode_in_between_roles and y[y_i] != y[y_i_paired]:
-            conf_vnode = rules.rules[winners[y_i]].stats.conf
-            conf_vnode_paired = rules.rules[winners[y_i_paired]].stats.conf
-            winner, loser = ((y_i, y_i_paired)
-                             if conf_vnode > conf_vnode_paired
-                             else (y_i_paired, y_i))
-            processed_y[loser] = y[winner]
-            processed_winners[loser] = winners[winner]
+        y_i_1 = y_indices[id(vnode1)]
+        y_i_3 = y_indices[id(vnode3)]
+        conf_vnode1 = rules.rules[winners[y_i_1]].stats.conf
+        conf_vnode3 = rules.rules[winners[y_i_3]].stats.conf
+        labels1 = tuple(feature_extractor.composite_to_labels[y[y_i_1]])
```

Comment:
We can make them lists and avoid the conversions below. At least I don't see why not.